k2/python/csrc/torch/pruned_ranges_to_lattice.cu

```diff
@@ -0,0 +1,282 @@
+ the information it represents.
+
+ When this is implemented, the lattice is used to generate force-alignment.
+ Perhaps you could find other uses for this function.
+
+   @param ranges  A tensor containing the symbol indexes for each frame.
+       Its shape is (B, T, s_range). See the docs in `get_rnnt_prune_ranges`
+       in k2/python/k2/rnnt_loss.py for more details of this tensor.
+       Its type is int32, consistent with that in rnnt_loss.py.
+   @param frames  The number of frames per sample with shape (B).
+       Its type is int32.
+   @param symbols  The symbol sequence, a LongTensor of shape (B, S),
+       and elements in {0..C-1}.
+       Its type is int64(Long), consistent with that in rnnt_loss.py.
+   @param logits  The pruned joiner network (or am/lm)
+       of shape (B, T, s_range, C).
+       Its type can be float32, float64, float16. Though float32 is mainly
+       used, float64 and float16 are also supported for future use.
+   @param [out] arc_map  A map from arcs in generated lattice to global index
+       of logits, or -1 if the arc had no corresponding score in logits,
+       e.g. arc pointing to super final state.
+   @return  Return an FsaVec, which contains the generated lattice.
+*/
+FsaVec PrunedRangesToLattice(
+    torch::Tensor ranges,   // [B][T][s_range]
+    torch::Tensor frames,   // [B]
+    torch::Tensor symbols,  // [B][S]
+    torch::Tensor logits,   // [B][S][s_range][C]
+    Array1<int32_t> *arc_map) {
+
+    TORCH_CHECK(ranges.get_device() == frames.get_device());
+    TORCH_CHECK(ranges.get_device() == symbols.get_device());
+    TORCH_CHECK(ranges.get_device() == logits.get_device());
+
+    TORCH_CHECK(ranges.dim() == 3, "ranges should be 3-dimensional");
+    TORCH_CHECK(frames.dim() == 1, "frames should be 1-dimensional");
+    TORCH_CHECK(symbols.dim() == 2, "symbols should be 2-dimensional");
+    TORCH_CHECK(logits.dim() == 4, "logits should be 4-dimensional");
+
+    TORCH_CHECK(torch::kInt == ranges.scalar_type());
+    TORCH_CHECK(torch::kInt == frames.scalar_type());
+    TORCH_CHECK(torch::kLong == symbols.scalar_type());
+
+    ContextPtr context;
+    if (ranges.device().type() == torch::kCPU) {
+      context = GetCpuContext();
+    } else if (ranges.is_cuda()) {
+      context = GetCudaContext(ranges.device().index());
+    } else {
+      K2_LOG(FATAL) << "Unsupported device: " << ranges.device()
+                    << "\nOnly CPU and CUDA are verified";
+    }
+
+    // "_a" is short for accessor.
+    auto ranges_a = ranges.accessor<int32_t, 3>();
+    auto frames_a = frames.accessor<int32_t, 1>();
+    auto symbols_a = symbols.accessor<int64_t, 2>();
+
+    // Typically, s_range is 5.
+    const int32_t B = ranges.size(0),
+                  T = ranges.size(1),
+                  s_range = ranges.size(2);
+
+    // Compute f2s_shape: fsa_to_state_shape.
+    Array1<int32_t> f2s_row_splits(context, B + 1);
+    int32_t * f2s_row_splits_data = f2s_row_splits.Data();
```

Comment:
```
int32_t * f2s_row_splits_data = f2s_row_splits.Data();
```

to be consistent, please replace 
`int32_t * f2s_row_splits_data` 
with
`int32_t *f2s_row_splits_data`.